tests/ui/traits/trait-upcasting/migrate-lint-deny.stderr

```diff
@@ -9,6 +9,7 @@ LL |     type Target = dyn A;
    |
    = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
```

Comment:
we should use `FutureIncompatibilityReason::FutureReleaseSemanticsChange` instead of `FutureReleaseErrorDontReportInDeps` for this lint. Can you change that in this PR?